db/view/view.cc

```diff
@@ -704,7 +704,8 @@ static const column_definition* view_column(const schema& base, const schema& vi
     // FIXME: Map base column_ids to view_column_ids, which can be something like
     // a boost::small_vector where the position is the base column_id, and the
     // value is either empty or the view's column_id.
-    return view.get_column_definition(base.column_at(kind, base_id).name());
+    const auto& col = base.column_at(kind, base_id);
+    return col.is_internal() ? nullptr : view.get_column_definition(col.name());
```

Comment:
As far as I understand the current state of the code, this change mostly affects `add_cells_to_view` and just prevents the internal column from being set in the view update mutation being generated here. There is also a piece of code in `table::do_push_view_replica_updates` which, in case of some materialized views, does a read-before-write and reads _all_ columns of the row being updated. We fetch all columns in order to calculate the lifetime of the row, but I'm not sure we should include internal columns there - both because they shouldn't be needed for liveness calculation and because it's slightly wasteful.

Please make sure to update the read-before-write code. Would be great to add a test, but I'm not sure it's possible to test materialized views with the boost framework.

Reply:
It's definitely possible to test MV with the boost framework (we have view_schema_test.cc and several others), but if you add a feature that adds internal columns you could also test it in Python.

Reply:
I think the view code only needs to be changed in ONE place - the place creating the view schema. Once the view schema doesn't contain a certain column X, its value is *not* copied into the view updates. You don't need to modify any of the view update code, it is already correct for the case that not all base columns are SELECTed into the view.